rhombus/src/internal/routes/scoreboard.rs

```diff
@@ -15,17 +15,36 @@ pub async fn route_scoreboard(
     user: Extension<MaybeUser>,
     page: Extension<PageMeta>,
     params: Query<PageParams>,
-    req: Request<Body>,
+    uri: Uri,
 ) -> impl IntoResponse {
     let challenge_data = state.db.get_challenges().await.unwrap();
     let default_division = challenge_data.divisions.keys().next().unwrap();
     if challenge_data.divisions.len() == 1 {
-        return route_scoreboard_division(state, user, page, Path(*default_division), params, req)
-            .await
-            .into_response();
+        return route_scoreboard_division(
+            state,
+            user,
+            page,
+            Path(default_division.to_string()),
+            params,
+            uri,
+        )
+        .await
+        .into_response();
     }
 
-    Redirect::temporary(format!("/scoreboard/{}", default_division).as_str()).into_response()
+    Redirect::temporary(
+        format!(
+            "/scoreboard/{}{}",
+            default_division,
+            if uri.path().ends_with(".json") {
+                ".json"
+            } else {
+                ""
+            }
+        )
+        .as_str(),
+    )
+    .into_response()
```

Comment:
This codepath is only hit in the UI, and it exists to redirect the user to the correct scoreboard page. The scoreboard js will fetch from `window.location.pathname + ".json"` (with your changes for the `+ .json`), which will always include the division id.

There is no need to for `/scoreboard.json` route or this special handling.

Reply:
What about when there's only one division. Before this PR `/scoreboard` was a special case where it wouldn't redirect but just send the correct response for the unique division.

Reply:
Good point. I think then these changes look good. I like the single division special case, since that is the most common deployment for most CTFs